OrbitCore/ConnectionManager.cpp

```diff
@@ -163,6 +164,18 @@ void ConnectionManager::SetupClientCallbacks()
         GCoreApp->RefreshCaptureView();
     } );
 
+    GTcpClient->AddMainThreadCallback ( Msg_SamplingCallstack, [=]( const Message& a_Msg )
```

Comment:
With a high volume of samplingCallstacks, this code will likely cause the UI thread to block making it hard to stop the capture or navigate the app while capturing.  "AddMainThreadCallback" was added for easy thread safety of small low frequency callbacks.  In this case, we should use "AddCallback" and make sure ProcessSamplingCallstack() itself is thread safe.

Reply:
Okay, I changed it to use `AddCallback` and made `AddCallstackEvent` thread-safe by using the already existing symbol mutex.